Access emulator database from Java client library

https://github.com/j143/query-spanner-with-java/blob/c85902d722188f83a3e2fc244c51e1b68f7356a7/.github/workflows/spanner-test.yml#L79-L86

The solution should look like,

```sh

- run: java -jar target/spanner-google-cloud-samples-jar-with-dependencies.jar  createdatabase test-instance example-db
  env:
      SPANNER_EMULATOR_HOST: localhost:9010
```

ERROR: (gcloud.spanner.databases.execute-sql) HTTPError 400: {"error":"Statement not supported: CreateTableStmt [at 1:1]\nCREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, P...\n^","code":3,"message":"Statement not supported: CreateTableStmt [at 1:1]\nCREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, P...\n^"}
Error: Process completed with exit code 1.

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -102,7 +102,19 @@ jobs:
       # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/create
       - name: Create database
         run: gcloud spanner databases create demo --instance=cloudspanner-lab
-        
+      
+      # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/execute-sql 
+      - name: Create Table
+        run: |
+          gcloud spanner databases execute-sql demo \
+            --instance=cloudspanner-lab  \
+            --sql='CREATE TABLE Songs {
+                     SongId INT64 NOT NULL,
+                     Title STRING(1024),
+                     Year INT64,
+                     Peak INT64
+                   } PRIMARY KEY (SongId);' \
+          
         
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar
         env:

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -108,7 +108,7 @@ jobs:
       # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/ddl/update
       - name: Create Table
         run: |
-          gcloud spanner databases execute-sql demo \
+          gcloud spanner databases ddl update demo \
             --instance=cloudspanner-lab  \
             --ddl='CREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, Peak INT64 } PRIMARY KEY (SongId);'
           

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -93,6 +93,7 @@ jobs:
           gcloud config set api_endpoint_overrides/spanner http://localhost:9020/
           
       - run: gcloud spanner instances create cloudspanner-lab --config=emulator-config --description="Test Instance"  --nodes=1
+      - run: gcloud spanner instances list
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar
         env:
           SPANNER_EMULATOR_HOST: localhost:9010

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -91,9 +91,19 @@ jobs:
           gcloud config set auth/disable_credentials true
           gcloud config set project spanner-lab-project
           gcloud config set api_endpoint_overrides/spanner http://localhost:9020/
-          
-      - run: gcloud spanner instances create cloudspanner-lab --config=emulator-config --description="Test Instance"  --nodes=1
-      - run: gcloud spanner instances list
+      
+      
+      - name: Create Spanner instance
+        run: |
+          gcloud spanner instances create cloudspanner-lab --config=emulator-config --description="Test Instance"  --nodes=1
+          gcloud spanner instances list
+          gcloud spanner instances describe cloudspanner-lab
+      
+      # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/create
+      - name: Create database
+        run: gcloud spanner databases create demo --instance=cloudspanner-lab
+        
+        
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar
         env:
           SPANNER_EMULATOR_HOST: localhost:9010

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -93,7 +93,7 @@ jobs:
           gcloud config set api_endpoint_overrides/spanner http://localhost:9020/
           
       - run: gcloud spanner instances create cloudspanner-lab --config=emulator-config --description="Test Instance"  --nodes=1
-      - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar  createdatabase cloudspanner-lab demo
+      - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar
         env:
           SPANNER_EMULATOR_HOST: localhost:9010
       

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -84,4 +84,7 @@ jobs:
           gcloud config set api_endpoint_overrides/spanner http://localhost:9020/
           
       - run: gcloud spanner instances create cloudspanner-lab --config=emulator-config --description="Test Instance"  --nodes=1
+      - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar  createdatabase cloudspanner-lab demo
+        env:
+          SPANNER_EMULATOR_HOST: localhost:9010
       

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -106,11 +106,14 @@ jobs:
       # Spanner Database docs:
       # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/execute-sql 
       # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/ddl/update
+      # 
+      # Data Definition Language (DDL) reference:
+      # https://cloud.google.com/spanner/docs/data-definition-language#naming_conventions
       - name: Create Table
         run: |
           gcloud spanner databases ddl update demo \
             --instance=cloudspanner-lab  \
-            --ddl='CREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, Peak INT64 } PRIMARY KEY (SongId);'
+            --ddl='CREATE TABLE Songs ( SongId STRING(1024) NOT NULL, Title STRING(1024), Year INT64, Peak INT64 ) PRIMARY KEY (SongId);'
           
         
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -39,6 +39,15 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - name: Checkout code
         uses: actions/checkout@v2
+      
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      
+      - name: Build with Maven
+        run: mvn -B package --file spanner-with-java/pom.xml
         
       - name: Set up gcloud Cloud SDK environment
         # You may pin to the exact commit or the version.

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -103,12 +103,14 @@ jobs:
       - name: Create database
         run: gcloud spanner databases create demo --instance=cloudspanner-lab
       
+      # Spanner Database docs:
       # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/execute-sql 
+      # https://cloud.google.com/sdk/gcloud/reference/spanner/databases/ddl/update
       - name: Create Table
         run: |
           gcloud spanner databases execute-sql demo \
             --instance=cloudspanner-lab  \
-            --sql='CREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, Peak INT64 } PRIMARY KEY (SongId);'
+            --ddl='CREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, Peak INT64 } PRIMARY KEY (SongId);'
           
         
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar

--- a/.github/workflows/spanner-test.yml
+++ b/.github/workflows/spanner-test.yml
@@ -108,12 +108,7 @@ jobs:
         run: |
           gcloud spanner databases execute-sql demo \
             --instance=cloudspanner-lab  \
-            --sql='CREATE TABLE Songs {
-                     SongId INT64 NOT NULL,
-                     Title STRING(1024),
-                     Year INT64,
-                     Peak INT64
-                   } PRIMARY KEY (SongId);' \
+            --sql='CREATE TABLE Songs { SongId INT64 NOT NULL, Title STRING(1024), Year INT64, Peak INT64 } PRIMARY KEY (SongId);'
           
         
       - run: java -jar spanner-with-java/target/spanner-with-java-1.0-SNAPSHOT-jar-with-dependencies.jar

--- a/spanner-with-java/src/main/java/page/janardhan/labs/App.java
+++ b/spanner-with-java/src/main/java/page/janardhan/labs/App.java
@@ -107,7 +107,7 @@ public class App {
                     .executeQuery(Statement.of("SELECT SongId, Title, Year FROM Songs"))) {
                 while (resultSet.next()) {
                     System.out.printf(
-                            "%d %d %s\n", resultSet.getString(0), resultSet.getString(1), resultSet.getLong(2));
+                            "%s %s %d\n", resultSet.getString(0), resultSet.getString(1), resultSet.getLong(2));
                 }
             }
         } finally {

